ktor-features/ktor-auth/src/io/ktor/auth/Headers.kt

```diff
@@ -78,8 +147,16 @@ sealed class HttpAuthHeader(val authScheme: String) {
             }
         }
 
+        /**
+         * Copies this [Parameterized] appending a new parameter [name] [value].
+         */
         fun withParameter(name: String, value: String) = Parameterized(authScheme, this.parameters + HeaderValueParam(name, value))
```

Comment:
I guess this should preserve the `encoding`. I can add another commit fixing this + test if confirmed.

Reply:
Yes, you are right, encoding is lost here